xarray/core/formatting.py

```diff
@@ -408,10 +408,24 @@ def coords_repr(coords, col_width=None, max_rows=None):
     )
 
 
-def summarize_index(
-    name: Hashable, index, col_width: int, max_width: int = None, is_index: bool = False
-):
-    return pretty_print(f"    {name} ", col_width) + f"{repr(index)}"
+def inline_index_repr(index, max_width=None):
+    if hasattr(index, "_repr_inline_"):
+        repr_ = index._repr_inline_(max_width=max_width)
+    else:
+        repr_ = repr(index)
```

Comment:
Could you add a small comment explaining that this line is only for the case of `pd.Index`?

While testing #7185 I was wondering why the full repr was not shown inline for the `CustomIndex` example.

, this will never be executed?

Reply:
of course!